File: examenes/ExamenDialogos/app/src/main/java/com/system/develop/examendialogos/DialogoConfirmacion.java
package com.system.develop.examendialogos;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.util.Log;

public class DialogoConfirmacion extends DialogFragment {

    OnConfirmacionListener listener;


    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            listener = (OnConfirmacionListener) context;
        }catch (ClassCastException e){
            Log.v("log","Error al castear el escuchador");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("Examen PMDM");
        builder.setMessage("Bienvenido al examen de PMDM, ¿estás seguro que quiere continuar?");
        builder.setPositiveButton("SI", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                listener.onConfirmacionSelected();
            }
        });
        builder.setNegativeButton("NO", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {

            }
        });

        return builder.create();
    }

    public interface OnConfirmacionListener{
        public void onConfirmacionSelected();
    }

}

File: Kotlin/T5/codigos2223/T5_Dalogos_RET/app/src/main/java/com/develop/t5_dalogos_ret/dialogs/DialogoConfirmacion.kt
package com.develop.t5_dalogos_ret.dialogs

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.develop.t5_dalogos_ret.R

class DialogoConfirmacion : DialogFragment() {

    private lateinit var contexto: Context
    private lateinit var listener: OnDialogoConfirmListener

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.contexto = context
        this.listener = context as OnDialogoConfirmListener
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        // AlertDialgo.Builder
        //val builder = AlertDialog.Builder(requireContext())
        val builder = AlertDialog.Builder(contexto)
        builder.setTitle("Continuar")
        //builder.setTitle(getString(R.string.titulo_dialogo_confirm))
        builder.setMessage("¿Estas seguro que quieres continuar?")

        // botones -- Positivo negativo neutral
        builder.setPositiveButton("OK", DialogInterface.OnClickListener { _,_ ->
            Log.v("dialogo","PULSADO OK")
            listener.onDialogoSelected("Seleccionado OK")
        })

        builder.setNegativeButton("CANCEL", DialogInterface.OnClickListener { _,_ ->
            Log.v("dialogo","PULSADO CANCELAR")
            listener.onDialogoSelected("Seleccionado CANCEL")
        })

        builder.setNeutralButton("SALIR", DialogInterface.OnClickListener { _,_ ->
            Log.v("dialogo","PULSADO SALIR")
        })
        return builder.create()
    }

    interface OnDialogoConfirmListener{
        fun onDialogoSelected(comunicacion: String)
    }
}